bundles/org.eclipse.cdt.lsp.test/src/org/eclipse/cdt/lsp/test/server/enable/HasLanguageServerPropertyTesterTest.java

```diff
@@ -0,0 +1,99 @@
+import org.eclipse.cdt.lsp.test.TestUtils;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.io.TempDir;
+
+class HasLanguageServerPropertyTesterTest {
+	private static final String EXTERNAL_HEADER_HPP = "ExternalHeader.hpp";
+	private IProject project;
+	
+	@TempDir
+	private static File TEMP_DIR;
+	
+	@BeforeEach
+	public void setUp(TestInfo testInfo) throws CoreException {
+		project = TestUtils.createCProject(TestUtils.getName(testInfo));
+	}
+	
+	@AfterEach
+	public void cleanUp() throws CoreException {
+		TestUtils.deleteProject(project);
+	}
+
+	/**
+	 * Tests whether LS enable test returns false for an external file when no (LSP) editor is opened.
+	 * @throws IOException 
+	 */
+	@Test
+	public void testLsNotEnabledForExternalFile_NoEditorOpen() throws CoreException, IOException {
+		//GIVEN is an external file which is not opened:
+		File externalFile = new File(TEMP_DIR, EXTERNAL_HEADER_HPP);
+		//WHEN the file is not opened,
+		//THEN the hasLanguageServerPropertyTester.test returns FALSE for the given file URI:
+		assertTrue(!new HasLanguageServerPropertyTester().test(externalFile.toURI(), null, null, null));	
+		//ensure clean up
+		externalFile.delete();
+	}
+	
+	/**
+	 * Tests whether LS enable test returns false for an external file which is opened in the C/C++ Editor.
+	 */	
+	@Test
+	public void testLsNotEnabledForExternalFile_OpenedInCEditor() throws CoreException, IOException {
+		//GIVEN is an existing external file:
+		File externalFile = new File(TEMP_DIR, EXTERNAL_HEADER_HPP);
+		externalFile.createNewFile();
+		//WHEN it's opened in the C/C++ Editor:
+		var editor = TestUtils.openInEditor(externalFile.toURI(), LspPlugin.C_EDITOR_ID);
+		//THEN the hasLanguageServerPropertyTester.test returns FALSE for the given file URI:
+		assertTrue(!new HasLanguageServerPropertyTester().test(externalFile.toURI(), null, null, null));
+
+		//ensure clean up
+		TestUtils.closeEditor(editor, false);
+		externalFile.delete();
+	}
+	
+	/**
+	 * Tests whether LS enable test returns true for an external file which is opened in the C/C++ Editor (LSP).
+	 */	
+	@Test
+	public void testLsEnableForExternalFile_OpenedInLspCEditor() throws CoreException, IOException {
+		//GIVEN is an existing external file:
+		File externalFile = new File(TEMP_DIR, EXTERNAL_HEADER_HPP);
+		externalFile.createNewFile();
+		//WHEN it's opened in the LSP based C/C++ Editor:
+		var editor = TestUtils.openInEditor(externalFile.toURI(), LspPlugin.LSP_C_EDITOR_ID);
+		//THEN the hasLanguageServerPropertyTester.test returns TRUE for the given file URI:
+		assertTrue(new HasLanguageServerPropertyTester().test(externalFile.toURI(), null, null, null));	
+
+		//ensure clean up
+		TestUtils.closeEditor(editor, false);
```

Comment:
Perhaps, this part wants to go to `@AfterEach`, while "create & open" counterpart tends to `@BeforeEach`